Firestore/core/src/firebase/firestore/api/firestore.mm

```diff
@@ -143,6 +144,25 @@
   [client_ shutdownWithCallback:std::move(callback)];
 }
 
+void Firestore::ClearPersistence(util::StatusCallback callback) {
+  worker_queue()->Enqueue([this, callback] {
+    std::lock_guard<std::mutex> lock{mutex_};
+    util::Status status;
+    if (!client_ || client().isShutdown) {
+      DatabaseInfo database_info(database_id_, persistence_key_,
```

Comment:
Reassembling the `DatabaseInfo` here is prone to error. Since we now need it here, we might as well have the `Firestore` constructor take it and store it directly.